src/rust/lib/frp/src/lib.rs

```diff
@@ -308,4 +308,47 @@ mod dynamic_mode_tests {
         some_event.emit(());
         assert_eq!(passed_events.get(), 2);
     }
+
+    #[test]
+    fn test_filter() {
+        let passed_events = Rc::new(Cell::new(0));
+        frp::new_network! { network
+            def source   = source::<bool>();
+            def filtered = source.filter(|&value| value == true);
+            eval filtered ([passed_events](&value) {
+                assert_eq!(value,true);
+                passed_events.set(passed_events.get() + 1);
+            });
+        };
+
+        source.emit(false);
+        source.emit(true);
+        source.emit(true);
+        source.emit(false);
+        source.emit(false);
+        assert_eq!(passed_events.get(),2);
+    }
+
+    #[test]
+    fn test_filter_map() {
+        let passed_events = Rc::new(Cell::new(0));
+        frp::new_network! { network
+            def source        = source::<bool>();
+            def filter_mapped = source.filter_map(|&value|
+                match value {
+                    true  => Some(()),
+                    false => None,
+                });
+            eval filter_mapped ([passed_events](_) {
+                passed_events.set(passed_events.get() + 1);
+            });
+        };
+
+        source.emit(false);
+        source.emit(true);
+        source.emit(true);
+        source.emit(false);
+        source.emit(false);
+        assert_eq!(passed_events.get(),2);
```

Comment:
Instead of hardcoding 2, compute it from the `input` variable proposed above. This will be less magical. Apply to the other test as well pls.